docs/DeveloperGuide.md

```diff
@@ -1,22 +1,264 @@
+- **Input Validation in `checkAnswer`**: By ensuring only non-null, non-empty strings are accepted as answers, `checkAnswer` reduces the risk of invalid input and enhances robustness.
+- **Placeholder in Question Text**: Using a placeholder (e.g., `"_____"`) in the question text makes it clear to users where they should mentally "fill in the blank," improving usability.
+
+#### Alternative Considerations
+
+1. **Implementing Case-Sensitive Comparison**: An initial approach was to use case-sensitive comparison for `checkAnswer`, but it was modified to ignore case for a more forgiving user experience.
+2. **Approximate Matching for Answers**: To increase flexibility, an alternative approach could involve approximate matching, allowing answers that closely resemble the correct answer to be accepted (e.g., `"paris"` vs. `"Paris"`). This would, however, add complexity to the validation process and was ultimately deemed unnecessary for this basic implementation.
+
+#### UML Class Diagram
+
+![FillInTheBlank UML Class Diagram](https://github.com/YubotKwng/tp/blob/e3358e1cc869012c202864071fbb75f2f8261712/docs/UML/FITBClassDiagram.png)
+
+The diagram illustrates the `FillInTheBlank` class with its attributes and methods, and its inheritance relationship with the `Question` superclass. This setup allows `FillInTheBlank` to share common functionality with other question types while implementing unique behavior specific to fill-in-the-blank questions.
+
+### Topic Class Implementation
+The `Topic` class is responsible for organizing questions under specific programming topics in the quiz application. It allows for modular storage of questions, making it easy for users to select topics and for developers to manage quiz content.
+
+#### **Feature Overview**
+The `Topic` class encapsulates a programming topic along with its related questions. It helps in grouping relevant questions, enabling quizzes to be topic-based, and provides essential methods to add and access questions within each topic.
+
+#### **Implementation Details**
+
+1. **Attributes**:
+    - `String name`: Stores the name of the topic.
+    - `List<Question> questions`: Holds a list of questions related to this topic.
+
+2. **Constructor**:
+    - `Topic(String name)`:  
+      Initializes the topic with a specified name and creates an empty list to hold its questions. Ensures that the `name` is non-null to prevent invalid topics.
+
+3. **Methods**:
+    - `addQuestion(Question question)`:  
+      Adds a `Question` object to the topic’s question list.
+    - `List<Question> getQuestions()`:  
+      Returns the list of questions under the topic.
+    - `String getName()`:  
+      Returns the name of the topic as a string.
```

Comment:
Is this really necessary? I feel that the implementation details becomes rather lengthy and repetitive as these are already being showcased in your class diagrams.